test/configparser.cpp

```diff
@@ -24,58 +26,110 @@ class DummyConfigHandler : public ConfigActionHandler {
 	std::vector<std::pair<std::string, std::vector<std::string>>> history;
 };
 
-TEST_CASE("parse_line(): Handles both lines with and without quoting",
+} // Anonymous namespace
+
+TEST_CASE("parse_line() Handles both lines with and without quoting",
 	"[ConfigParser]")
 {
 	ConfigParser cfgParser;
 
-	DummyConfigHandler handler;
+	ConfigHandlerHistoryDummy handler;
 	cfgParser.register_handler("command-name", handler);
 
 	const std::string location = "dummy-location";
 
-	SECTION("no quoting") {
-		cfgParser.parse_line(" command-name ", location);
-		cfgParser.parse_line(" command-name arg1 ", location);
-		cfgParser.parse_line(" command-name arg1 arg2 ", location);
+	SECTION("unknown command results in exception") {
+		REQUIRE_THROWS(cfgParser.parse_line("foo", location));
+		REQUIRE_THROWS(cfgParser.parse_line("foo arg1 arg2", location));
+	}
 
-		REQUIRE(handler.history.size() == 3);
-		REQUIRE(handler.history[0].second == std::vector<std::string>({}));
-		REQUIRE(handler.history[1].second == std::vector<std::string>({"arg1"}));
-		REQUIRE(handler.history[2].second == std::vector<std::string>({"arg1", "arg2"}));
+	SECTION("different combinations of (un)quoted commands/arguments have the same result") {
+		std::vector<std::string> inputs = {
```

Comment:
Can this (and other vectors below) be const? :P

Reply:
Done while rebasing/squashing: https://github.com/newsboat/newsboat/compare/de0dd8f03cdf6b8c14f04e9bd33a57dec4744d13..de634778c3d36f6356f93cc764fbc96c7edff345

Reply:
Thanks!